pytest_watch/watcher.py

```diff
@@ -109,26 +110,34 @@ def run(self, summary=None):
 
         # Run py.test or py.test runner
         exit_code = subprocess.call(argv, shell=(sys.platform == 'win32'))
-        # py.test returns exit code 5 in case no tests are run/collected.
-        # This can happen with tools like pytest-testmon.
-        passed = exit_code == 0 or (runner == 'py.test' and exit_code == 5)
+
+        if runner == 'py.test':
+            # py.test returns exit code 5 in case no tests are run/collected.
+            # This can happen with tools like pytest-testmon.
+            passed = exit_code in [0, 5]
+        else:
+            passed = exit_code == 0
+        failed = not passed
+        interrupted = exit_code == 2
 
         # Beep if failed
-        if not passed and self.beep_on_failure:
+        if failed and self.beep_on_failure:
             sys.stdout.write(BEEP_CHARACTER)
             sys.stdout.flush()
 
         # Run custom commands
+        if interrupted and self.onsigint:
+            os.system(self.onsigint)
```

Comment:
This is never getting called for me, since pressing `CTRL+C` raises an exception in ptw, bubbling up to the outer `KeyboardInterrupt` handler [below](https://github.com/joeyespo/pytest-watch/pull/46/files#diff-298efc967591604e348416060d077075R174).

Is this code being executed for you?